src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,136 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_CAPACITY_COEFFICIENT = 1.5;
+    private Object[] data;
+    private int size;
+
+    public ArrayList() {
+        this.data = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= data.length) {
+            data = grow();
+        }
+        data[size] = value;
+        size += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index "
+                    + index
+                    + " is out of bounds for the size "
+                    + (size));
+        }
+        isIndexPositiveNumber(index);
+        if (data.length < size + 1) {
+            data = grow();
+        }
+        System.arraycopy(data, index, data, index + 1, size - index);
+        data[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int newSize = this.size + list.size();
+        if (data.length < newSize) {
+            data = grow(newSize);
+        }
+        for (int i = size, j = 0; i < newSize; i++, j++) {
+            data[i] = list.get(j);
+            size++;
```

Comment:
The `size` should be updated once after the loop in the `addAll` method, not during each iteration. Move `size = newSize;` after the loop.